### ISSUE TITLE
Config key to disable GraphQL

### ISSUE THREAD

**Issue by wolfgang42** [holidayextras/jsonapi-server#256]
#233 introduced the ability to disable GraphQL by passing the option `graphiql: false`. (One example of why this option is useful is #244: there is reported a problem with an interaction between `joi.object()` and GraphQL. I have also encountered this problem, but for the moment do not need GraphQL support and so have turned it off.)

In https://github.com/holidayextras/jsonapi-server/pull/233#issuecomment-269498049 @theninj4 suggested that this option would be better as a separate key (since it controls not just Graph*i*QL but all of GraphQL), and proposed the name `strict: false`, but I have found no further discussion on the topic. I personally find `strict` to be an odd name for it, and would propose instead `graphql: false` (with a default of `true`).

**Comment by st3xupery** [comment]
Forgive me, but I am a little confused by what you are proposing then is different than the way the setting is handled now by the PR resulting from #223. graphiql:false is not the default but will turn off graphiql.

**Comment by jessevarnado** [comment]
I agree with @wolfgang42 that we need to be able to toggle GraphQL support independently from GraphiQL support. Per the suggestion in [express-graphql](https://github.com/graphql/express-graphql#options), `graphiql` should be turned off in environments other than development. My project makes use of jsonapi-server's GraphQL functionality in all our environments, but we would like to turn off support for GraphiQL in non-development environments.

My suggestion would be to support a configuration property named `graphql` that could contain an object specifying the configuration options provided by [express-graphql](https://github.com/graphql/express-graphql#options). For example: 
```javascript
var jsonApi = require("jsonapi-server");

jsonApi.setConfig({
  port: 16006,
  graphql: {
    graphiql: false,
    pretty: true,
    ...
  }
});
```

If you would like to turn off GraphQL support completely, then you would simply set `graphql: false`.

**Comment by vnen** [comment]
Any progress on this? I think it's very important to have those options separated. I want to use GraphQL on the server, but Graph*i*QL should be disabled on production.

**Comment by pmcnr-hx** [comment]
Thank you for the reminder @vnen. I'll merge later today.

**Comment by pmcnr-hx** [comment]
I concur that GraphiQL should be disabled in production. I'll look into this again today @vnen.

**Comment by wolfgang42** [comment]
@pmcnr-hx Did you ever have a chance to look into this?

**Comment by paparomeo** [comment]
@wolfgang42: it's been a busy week but I'll definitely have a look at this until the end of the week.
